packages/fxa-auth-server/lib/l10n/index.ts

```diff
@@ -91,26 +95,35 @@ class Localizer {
    * @param locale Locale to use, defaults to en.
    */
   protected async fetchTranslatedMessages(locale?: string) {
-    // note: 'en' auth.ftl only exists for browser bindings / Storybook
-    // the fallback English strings within the templates will be shown in other envs
+    // Note: 'en' auth.ftl only exists for browser bindings / Storybook. The fallback
+    // English strings within the templates are tested and are shown in other envs
     const path = `${this.bindings.opts.translations.basePath}/${
       locale || 'en'
     }/auth.ftl`;
 
     return this.bindings.fetchResource(path);
   }
 
-  // NOTE: not currently used, will be implemented in FXA-4623
-  protected async localizeString(
-    acceptLanguage: string,
-    { id, message }: FtlIdMsg
-  ) {
+  async localizeStrings(
+    acceptLanguage = 'en',
+    ftlIdMsgs: FtlIdMsg[]
+  ): Promise<LocalizedStrings> {
     const { l10n } = await this.setupLocalizer(acceptLanguage);
```

Comment:
I am not 100% sure, but this call might be heavier than we think. Considering this get's invoked from /session/verify/send_push, it'd be bad if this blocked the event loop in anyway. I wonder if we should double check the performance profile here. Let me know if you want to try an setup a preliminary load test.

Reply:
I'm trying to think of an alternative here. `setupLocalizer` only fetches the bundles from the `acceptLanguage`, so I hope it's not too heavy. 😩 I don't think we can instantiate `Localizer` at the top of files to then immediately fetch all of the bundles for all locales (since we wouldn't have access to `acceptLanguage` yet) since we'd still have to `await` for that to finish.

I wonder 👀 We have `acceptLanguage` in [a lazy getter](https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/lib/server.js#L201) on `app.request` so we can reference it with `request.app.acceptLanguage`. Maybe we can set up all of the Fluent bundles similarly? Would be worth checking for this kind of optimization in #12045 or even in #12191 since that may get rid of `translator` which `defineLazyGetter(request.app, 'locale'` uses.

I'm not sure in the meantime, without this kind of refactor, if there's a better way to do this. I'd be interested in what the performance profiler says either way, I keep referencing #12045 but I'd love to get a before vs. after that ticket.